app/services/providers/copy_to_recruitment_cycle_service.rb

```diff
@@ -1,8 +1,11 @@
 module Providers
   class CopyToRecruitmentCycleService
-    def initialize(copy_course_to_provider_service:, copy_site_to_provider_service:)
+    attr :logger
+
+    def initialize(copy_course_to_provider_service:, copy_site_to_provider_service:, logger: nil)
       @copy_course_to_provider_service = copy_course_to_provider_service
       @copy_site_to_provider_service = copy_site_to_provider_service
+      @logger = logger
```

Comment:
```suggest
      @logger = logger || Logger.new("/dev/null")
```

Reply:
You can use null object and remove the `if` statements

Reply:
will that work on a windows?

Reply:
> will that work on a windows?

It should running under mingw or WSL. :)